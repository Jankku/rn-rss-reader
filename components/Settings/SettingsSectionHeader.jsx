import { Text, StyleSheet } from 'react-native';
import { useTheme } from '@react-navigation/native';

function SettingsSectionHeader({ title }) {
  const { colors } = useTheme();
  const styles = makeStyles(colors);

  return <Text style={styles.title}>{title}</Text>;
}

const makeStyles = (colors) =>
  StyleSheet.create({
    title: {
      color: colors.text,
      fontSize: 22,
      fontWeight: '700',
      paddingHorizontal: 16,
      paddingVertical: 8,
    },
  });

export default SettingsSectionHeader;
