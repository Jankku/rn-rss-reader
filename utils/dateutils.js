import { DateTime } from 'luxon';

/**
 * For articles released today return time and date otherwise.
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date
 */
export function RFC2822ToTimeOrDate(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate =
    dateObject.toRelativeCalendar() === 'today'
      ? dateObject.toLocaleString(DateTime.TIME_24_SIMPLE)
      : dateObject.toLocaleString(DateTime.DATE_SHORT);
  return formattedDate;
}

/**
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date in DateTime format
 */
export function RFC2822ToDateTime(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate = dateObject.toLocaleString(DateTime.DATETIME_SHORT);
  return formattedDate;
}
