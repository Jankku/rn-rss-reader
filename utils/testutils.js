import { render } from '@testing-library/react-native';
import { ThemeContextProvider } from '../context/ThemeContext';
import { LocationContextProvider } from '../context/LocationContext';
import { RegionContextProvider } from '../context/RegionContext';

const AllProviders = ({ children }) => {
  return (
    <LocationContextProvider>
      <RegionContextProvider>
        <ThemeContextProvider>{children}</ThemeContextProvider>
      </RegionContextProvider>
    </LocationContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
