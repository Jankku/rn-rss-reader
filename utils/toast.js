import Toast from 'react-native-root-toast';

const showToast = (message) =>
  Toast.show(message, {
    position: 100,
    opacity: 0.9,
    shadow: false,
  });

export default showToast;
