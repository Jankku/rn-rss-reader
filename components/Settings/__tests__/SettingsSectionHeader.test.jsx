import SettingsSectionHeader from '../SettingsSectionHeader';
import { render } from '../../../utils/testutils';

describe('Test SettingsSectionHeader', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(<SettingsSectionHeader title={'About'} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
