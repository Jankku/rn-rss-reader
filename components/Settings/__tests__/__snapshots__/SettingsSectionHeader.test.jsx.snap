// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SettingsSectionHeader Should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#101010",
      "fontSize": 22,
      "fontWeight": "700",
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  About
</Text>
`;
